component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -2389,4 +2410,29 @@ private void setUserClaimValues(AuthenticatedUser authenticatedUser, Map<String,
                     String.format("Failed to update user claims for user : %s.", authenticatedUser), e);
         }
     }
+
+    private String getUnlockTime(AuthenticatedUser authenticatedUser)
+            throws AuthenticationFailedException {
+
+        String username = authenticatedUser.toFullQualifiedUsername();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        String unlockTime = null;
+        try {
+            UserRealm userRealm = getUserRealm(username);
+            if (userRealm == null) {
+                throw new AuthenticationFailedException("UserRealm is null for user : " + username);
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager != null) {
```

Comment:
Shall we change this logic to check `null` and add a debug log?

Reply:
fixed in be18eca